assignments/life_expectancy/cleaning.py

```diff
@@ -1,75 +1,69 @@
 """
 Functions to cleaning data
 """
-from typing import List
+from typing import List, Dict
 import time
 import logging as log
 import argparse
 import pandas as pd
 import numpy as np
-from life_expectancy.utils import f_input_configs, import_csv_to_pd, \
-    save_pd_to_csv, f_remove_non_digits
+from life_expectancy.utils import f_input_configs, import_csv_to_pd as load_data, \
+    save_pd_to_csv as save_data, f_remove_non_digits
 log.getLogger().setLevel(log.INFO)
 
 
-def clean_data(location_filter: str = 'PT') -> None:
+def clean_data(  # pylint: disable=too-many-arguments
+        df: pd.DataFrame,  # pylint: disable=invalid-name
+        rename_raw_cols: Dict[str, str],
+        unpivot_col_id_vars: List[str],
+        rename_unpivot_cols: Dict[str, str],
+        types_unpivot_cols: Dict[str, type],
+        representations_of_nan: List[str],
+        location_col: str,
+        location_filter: str
+) -> pd.DataFrame:
     """
-    Main Script to clean the data.
-
-    It extracts data from .csv, transforms and saves to a new .csv
-
-    Currently it only supports filtering by one region at a time.
-
-    :return: None
+    This function cleans and transforms the data
+
+    :param location_col: str
+        name of the col to filter location
+    :param representations_of_nan: list
+        the possible representations od NaN
+    :param types_unpivot_cols: dict
+        dict w/ types per col {col: type}
+    :param rename_unpivot_cols: dict
+        rename cols {old_name: new_name}
+    :param unpivot_col_id_vars: list
+        list w/ names of the columns used as id for unpivot
+    :param rename_raw_cols: dict
+        id columns that we want to rename {old_name: new_name}
+    :param df: pd.Dataframe
+        to clean and transform
+    :param location_filter: str
+        location filter
+    :return: pd.Dataframe
+        transformed
```

Comment:
This is nitpicking but the docstring would improve from having the parameters in the order that they appear previously

Reply:
True. I will change it. many tks. kudos!